packages/cli/src/util/telemetry/base.ts

```diff
@@ -7,4 +7,406 @@ export class TelemetryBaseClient extends TelemetryClient {
       value: actual,
     });
   }
+
+  trackCIVendorName() {
+    this.trackCI(getContinuousIntegrationVendorName());
+  }
+}
+
+interface Vendor {
+  readonly name: string;
+  readonly env: {
+    readonly any: string[];
+    readonly all: string[];
+  };
+  readonly eval_env: Record<string, string> | null;
 }
+
+export function getContinuousIntegrationVendorName() {
+  for (const env of VENDORS) {
+    if (env.eval_env) {
+      for (const [name, expected_value] of Object.entries(env.eval_env)) {
+        if (process.env[name] && process.env[name] === expected_value) {
+          return env.name;
+        }
+      }
+    } else if (env.env.any.length !== 0) {
+      for (const env_var of env.env.any) {
+        if (process.env[env_var]) {
+          return env.name;
+        }
+      }
+    } else if (env.env.all.length !== 0) {
+      if (env.env.all.every(env_var => Boolean(process.env[env_var]))) {
+        return env.name;
+      }
+    } else {
+      return undefined;
+    }
+  }
+}
+
+// list copied from Turborepo
+// https://github.com/vercel/turborepo/blob/663d3ffdc5b1a4a93353ed408c0b6a653fd89a9f/crates/turborepo-ci/src/vendors.rs#L33C13-L678C14
+export const VENDORS: Vendor[] = [
```

Comment:
We may want to consider using [`ci-info`](https://www.npmjs.com/package/ci-info) package instead of maintaining this ourselves.

Reply:
After your suggestion, @trek and I checked out [`ci-info`](https://www.npmjs.com/package/ci-info). The main problem is that it decides whether you're in CI at package _import_ time, which makes it impossible to test well. 

In addition, the most useful feature, `ci.name` does not have a guarantee of a stable name, which means that we might arbitrarily start collecting a previous CI system under a new name without warning at some point in the future, making it difficult or impossible to compare historical and new data.

For these reasons, along with the relative small size of the CI checking function, I think it's worthwhile to retain a locally implemented, testable version of the logic.

Reply:
We saw that some other Vercelians have committed to the project though. We might be able to PR having stable identifiers and swap out our custom solution for the library if we can get a consistent value in the future.

Reply:
https://github.com/watson/ci-info/issues/130 we can see if the author is amenable